pwiz_tools/Skyline/SettingsUI/ViewLibraryPepInfoList.cs

```diff
@@ -31,7 +31,8 @@ namespace pwiz.Skyline.SettingsUI
 {
     public class ViewLibraryPepInfoList : AbstractReadOnlyList<ViewLibraryPepInfo>
     {
-        private readonly ImmutableList<ViewLibraryPepInfo> _allEntries;
+        private readonly ImmutableList<ViewLibraryPepInfo> _allEntries; // Alpha sorted
+        private readonly ImmutableList<int> _naturalSortIndex; // Natural sort order of the members of _allEntries
```

Comment:
Is there a reason that we don't want to just start off with having everything sorted naturally, instead of maintaining one ordering for the way things are initially displayed, and different one for when they apply a filter?

Reply:
It's about searching for partial matches while the user is typing in a filter string. We don't need or want anything fancier than alphanumeric matching for that, so the sort order is just straight up string comparison. And it's faster too.